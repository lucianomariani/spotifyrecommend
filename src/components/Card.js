import React from 'react'
import nextButton from '../assets/images/next.png'
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types'

class Card extends React.Component {

  render(){
    const { to, image, title, subtitle, text} = this.props
    let nextLink
    if (to) { nextLink = <img src={nextButton} height='32' className="next" alt="next"/>}
    let card = <div className="card">
                  <img src={image} alt={title} width='40%'/>
                    <div>
                      <h2>{title}</h2>
                      <h3>{subtitle}</h3>
                      <p>{text}</p>
                      { nextLink }
                    </div>
                </div>
    
    if (to) {
      card = <Link to={to}>{card}</Link>
    }
    return(<div>{card}</div>)
  }

}
export default Card

Card.propTypes = {
  to: PropTypes.string, 
  image: PropTypes.string.isRequired, 
  title: PropTypes.string.isRequired, 
  subtitle: PropTypes.string.isRequired,  
  text: PropTypes.string
};