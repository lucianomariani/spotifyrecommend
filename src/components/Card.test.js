import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Card from './Card'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter()});

const image = 'imagetest.jpg'
const title = 'title of card'
const subtitle = 'subtitle of card'


describe('Card ', () => {
  test('renders content', () => {
    const wrapper = shallow(<Card image={image} subtitle={subtitle} title={title}/>);
    expect(wrapper).toMatchSnapshot()
  });
});

