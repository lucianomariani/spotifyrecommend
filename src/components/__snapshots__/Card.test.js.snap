// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card  renders content 1`] = `ShallowWrapper {}`;
